src/net.c

```diff
@@ -1359,3 +1360,139 @@ void hyper_cleanup_dns(struct hyper_pod *pod)
 
 	close(fd);
 }
+
+int hyper_setup_iptables(struct hyper_pod *pod) {
+	int pid, status;
+	int pipefd[2] = {-1, -1};
+	int pipe_parent_err[2] = {-1, -1};
+	int ret = -1;
+	size_t size, offset = 0;
+	int pipe_size = 0;
+	const char *cmd = "iptables-restore";
+
+	if (! pod) {
+		return -1;
+	}
+
+	if (! pod->iptable_rules) {
+		return 0;
+	}
+
+	if (! *(pod->iptable_rules)) {
+		free(pod->iptable_rules);
+		return 0;
+	}
+
+	if (pipe(pipefd) == -1) {
```

Comment:
Sure, I only thought it would have been better to check 0, because in case there is a bug and it returns another error code than -1, we would continue the code sequence.